File: app/src/main/java/com/lumiyaviewer/lumiya/render/spatial/SpatialListEntry.java
package com.lumiyaviewer.lumiya.render.spatial;
import java.util.*;

public class SpatialListEntry<T> {
    public final T data;
    SpatialListEntry<T> next = null;
    SpatialTreeNode node = null;
    SpatialListEntry<T> prev = null;

    public SpatialListEntry(T t) {
        this.data = t;
    }

    public final SpatialListEntry<T> getNext() {
        return this.next;
    }
}

File: Linkpoint/src/main/kotlin/com/linkpoint/render_migrated/spatial/SpatialListEntry.kt
package com.linkpoint.render.spatial
import java.util.*

class SpatialListEntry<T> {
    val T data
    SpatialListEntry<T> next = null
    SpatialTreeNode node = null
    SpatialListEntry<T> prev = null

    public SpatialListEntry(T t) {
        this.data = t
    }

    val SpatialListEntry<T> getNext() {
        return this.next
    }
}
